packages/core/android/src/main/java/io/sentry/react/RNSentrySDK.java

```diff
@@ -0,0 +1,82 @@
+package io.sentry.react;
+
+import android.content.Context;
+import com.facebook.react.bridge.ReadableMap;
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import io.sentry.android.core.AndroidLogger;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
+
+public final class RNSentrySDK {
+  private static final String CONFIGURATION_FILE = "sentry.options.json";
+  private static final String NAME = "RNSentrySDK";
+
+  private static final ILogger logger = new AndroidLogger(NAME);
+  private static final RNSentryStart startSdk = new RNSentryStart();
+
+  private RNSentrySDK() {
+    throw new AssertionError("Utility class should not be instantiated");
+  }
+
+  private static void startWithOptions(
+      @NotNull final Context context, @NotNull final ReadableMap rnOptions) {
+    startSdk.startWithOptions(context, rnOptions, null, logger);
+  }
+
+  /**
+   * Start the Native Android SDK with the provided options
+   *
+   * @param context Android Context
+   * @param options Map with options
+   */
+  public static void startWithOptions(
+      @NotNull final Context context, @NotNull final Map<String, Object> options) {
```

Comment:
Let's match the `SentryAndroid` init signature, so the RN manual init can be a drop in replacement.

https://github.com/getsentry/sentry-java/blob/160762108453416271f7135b4e1d4dfc329fe871/sentry-android-core/src/main/java/io/sentry/android/core/SentryAndroid.java#L67-L77

Reply:
> This public interface should allow user to overwrite/extend the options specified in the sentry.options.json file using the configuration handler.

I think this is a great idea and makes the exposed api really powerful allowing for dynamic configurations that a json file cannot provide. Updated with 25d142a3ca9653c5da2cb4a8f98c7a29bab48309